lib/inventory/controller.go

```diff
@@ -596,6 +596,14 @@ func (c *Controller) handleSSHServerHB(handle *upstreamHandle, sshServer *types.
 	if sshServer.GetName() != handle.Hello().ServerID {
 		return trace.AccessDenied("incorrect ssh server ID (expected %q, got %q)", handle.Hello().ServerID, sshServer.GetName())
 	}
+	if hostname := sshServer.GetHostname(); hostname != "" {
+		// Allow uppercase node hostnames, they aren't valid hostnames
+		// but there's no reason to reject them here. The goal here is
+		// to disallow most special characters.
+		if !utils.IsValidHostname(strings.ToLower(hostname)) {
+			return trace.BadParameter("invalid hostname %q", hostname)
```

Comment:
nit: its helpful to be very explicit with inventory control stream errors as they can be bit tricky to debug.

```suggestion
			return trace.BadParameter("invalid ssh server hostname %q in heartbeat", hostname)
```

Reply:
Thanks, done